ed_sensor_integration/src/laser/updater.cpp

```diff
@@ -0,0 +1,655 @@
+
+    //pose_cache.clear();
+}
+
+void LaserUpdater::update(const ed::WorldModel& world, std::vector<double>& sensor_ranges,
+                         const geo::Pose3D& sensor_pose, const double timestamp, ed::UpdateRequest& req)
+{
+    tue::Timer t_total;
+    t_total.start();
+
+    uint num_beams = sensor_ranges.size();
+    // - - - - - - - - - - - - - - - - - -
+    // Filter measurment
+
+    for(unsigned int i = 1; i < num_beams - 1; ++i)
+    {
+        double rs = sensor_ranges[i];
+        // Get rid of points that are isolated from their neighbours
+        if (std::abs(rs - sensor_ranges[i - 1]) > 0.1 && std::abs(rs - sensor_ranges[i + 1]) > 0.1)  // TODO: magic number
+        {
+            sensor_ranges[i] = sensor_ranges[i - 1];
+        }
+    }
+
+    // - - - - - - - - - - - - - - - - - -
+    // Render world model as if seen by laser
+
+    std::vector<double> model_ranges(num_beams, 0);
+    renderWorld(sensor_pose, world, model_ranges);
+
+    // - - - - - - - - - - - - - - - - - -
+    // Fit the doors
+
+    geo::Pose3D sensor_pose_inv = sensor_pose.inverse();
+
+    if (fit_entities_)
+    {
+        std::cout << "Fitting!" << std::endl;
+
+        for(ed::WorldModel::const_iterator it = world.begin(); it != world.end(); ++it)
+        {
+            const ed::EntityConstPtr& e = *it;
+            //std::cout << e->type() << std::endl;
+
+            if (!e->shape() || !e->has_pose())
+                continue;
+
+            geo::Pose3D e_pose_SENSOR = sensor_pose_inv * e->pose();
+
+            // If not in sensor view, continue
+            if (e_pose_SENSOR.t.length2() > 5.0 * 5.0 || e_pose_SENSOR.t.x < 0)
+                continue;
+
+            if (e->hasType("left_door") || e->hasType("door_left") || e->hasType("right_door") || e->hasType("door_right"))
+            {
+                // Try to update the pose
+                geo::Pose3D new_pose = fitEntity(*e, sensor_pose, lrf_model_, sensor_ranges, model_ranges, 0, 0.1, 0, 0.1, -1.57, 1.57, 0.1, pose_cache);
+                req.setPose(e->id(), new_pose);
+
+                // Render the door with the updated pose
+                geo::LaserRangeFinder::RenderOptions opt;
+                opt.setMesh(e->shape()->getMesh(), sensor_pose_inv * new_pose);
+
+                geo::LaserRangeFinder::RenderResult res(model_ranges);
+                lrf_model_.render(opt, res);
+            }
+        }
+    }
+
+    // - - - - - - - - - - - - - - - - - -
+    // Try to associate sensor laser points to rendered model points, and filter out the associated ones
+
+    associate(sensor_ranges, model_ranges);
+
+    // - - - - - - - - - - - - - - - - - -
+    // Segment the remaining points into clusters
+
+    std::vector<ScanSegment> segments = segment(sensor_ranges);
+
+    // - - - - - - - - - - - - - - - - - -
+    // Convert the segments to convex hulls, and check for collisions with other convex hulls
+
+    std::vector<EntityUpdate> clusters(segments.size());
```

Comment:
Reduce the big comments to just single line comments. Also no need for empty lines between comment and code.